modules/irsa/data.tf

```diff
@@ -23,21 +23,13 @@ data "aws_iam_policy_document" "irsa_with_oidc" {
 
     principals {
       type        = "Federated"
-      identifiers = [local.eks_oidc_provider_arn]
+      identifiers = [var.eks_oidc_provider_arn]
     }
 
     condition {
       test     = "StringEquals"
-      variable = "${local.eks_oidc_issuer_url}:sub"
+      variable = "${var.eks_oidc_issuer_url}:sub"
```

Comment:
This should be `variable = "${replace(var.eks_oidc_issuer_url, "https://", "")}:sub"`